api/types/compare/equal.go

```diff
@@ -0,0 +1,22 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package compare
+
+// IsEqual[T] will be used instead of cmp.Equal if a resource implements it.
+type IsEqual[T any] interface {
+	IsEqual(T) bool
```

Comment:
I think there's an argument for `IsEqual(any) bool` - ie, a non-generic version that could do polymorphic equality. I have a feeling we talked about this before too.

(No changes expected here.)